src/main/java/com/powsybl/openloadflow/network/impl/LfAreaImpl.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2024, Coreso SA (https://www.coreso.eu/) and TSCNET Services GmbH (https://www.tscnet.eu/)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.network.impl;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.*;
+import com.powsybl.openloadflow.network.*;
+import com.powsybl.openloadflow.util.Evaluable;
+import com.powsybl.openloadflow.util.PerUnit;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+/**
+ * @author Valentin Mouradian {@literal <valentin.mouradian at artelys.com>}
+ */
+public class LfAreaImpl extends AbstractPropertyBag implements LfArea {
+
+    private final LfNetwork network;
+    private final Ref<Area> areaRef;
+
+    private final double interchangeTarget;
+
+    private final Set<LfBus> buses;
+
+    private Set<Supplier<Evaluable>> boundariesP;
+
+    private Map<LfBus, Double> externalBusesSlackParticipationFactors = new HashMap<>();
+
+    public LfAreaImpl(Area area, LfNetwork network, LfNetworkParameters parameters) {
```

Comment:
protected

Reply:
changed to protected in 9a9478c80af6e5e216bc90e464be48e7287c92c6